src/plugins/dataflow/nodes/dataflow-node.scss

```diff
@@ -276,24 +276,40 @@ $node-inner-width: 140px;
       .type-options span {
         border-top-color: $control-green;
       }
+      .label {
+        padding-left:3px;
+        text-align: left;
+        width: 100%;
+      }
     }
+    // OFF -- when not-holding, default
     &.has-gate .inputs {
-      div:first-of-type .socket {
-        transform:rotate(0deg);
-        background-image: url("../assets/icons/control/gate.svg");
-        transition: all .4s;
-        border-radius: 10px 4px 4px 10px;
+      // BINARY
+      div:nth-of-type(2) .socket {
+        background-image: url("../assets/icons/control/node-arrow.svg");
+        transform:rotate(270deg);
       }
     }
+    //  ON --- additional for when holding
     &.has-gate.gate-active .inputs {
-      div:first-of-type .socket {
+      // VALUE
+      div:nth-of-type(1) .socket {
+        background-image: url("../assets/icons/control/signal.svg");
+      }
+      // BINARY
+      div:nth-of-type(2) .socket {
```

Comment:
Using `nth-of-type` here seems like it'll cause confusion in the future. Is there any way we can use classes for this, or some other more explicit way of styling these divs? I'm imagining a future where we have to rearrange the order of these and then someone has to spend a couple hours figuring out why the icons aren't following the divs as expected. Maybe I'm over-worrying?

Reply:
I think that's a good idea.   I added in a dynamic class so we can always target sockets by the name they are passed in their constructor.   Those names are not always the most descriptive, but it's better than the nth-of-type.